src/storage/src/cell_serializer.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::Row;
+use risingwave_common::catalog::ColumnId;
+use risingwave_common::error::Result;
+
+pub type KeyBytes = Vec<u8>;
+pub type ValueBytes = Vec<u8>;
+
+pub trait CellSerializer {
```

Comment:
I am curious about the `pk` and `column_ids` in all three methods, will there be a case that
the same struct that implements `CellSerializer` may use different `pk` and `column_ids` when making function calls during its lifetime?

Reply:
I think only `pk` changes. Can't think of any case where `column_ids` needs to be changed. Even in `DedupPkCellBasedRowSerializer`, the deduplicated column ids should stay constant throughout its lifetime.

Reply:
> I think only pk changes. Can't think of any case where column_ids needs to be changed. Even in DedupPkCellBasedRowSerializer, the deduplicated column ids should stay constant throughout its lifetime.

Will refactor for this.

Reply:
Places which reference `CellSerializer::serialize` use `&self.column_ids` which doesn't change, but `pk` changes:
- https://github.com/singularity-data/risingwave/blob/8f0e0b24d8615287979ca080defb08a1dd231efd/src/storage/src/table/cell_based_table.rs#L224
- https://github.com/singularity-data/risingwave/blob/8f0e0b24d8615287979ca080defb08a1dd231efd/src/storage/src/table/cell_based_table.rs#L243

Reply:
Updated for cell_based_serializer to own column_ids, and do not require explicitly passing them. Will make another review pass tmr.

Reply:
Fixed.